internal/sdp/jsep.go

```diff
@@ -34,17 +34,13 @@ const (
 // some settings that are required by the JSEP spec.
 func NewJSEPSessionDescription(fingerprint string, identity bool) *SessionDescription {
 	d := &SessionDescription{
-		ProtocolVersion: 0,
-		Origin: fmt.Sprintf(
-			"- %d %d IN IP4 0.0.0.0",
-			newSessionID(),
-			time.Now().Unix(),
-		),
-		SessionName: "-",
-		Timing:      []string{"0 0"},
-		Attributes: []string{
-			//	"ice-options:trickle", // TODO: implement trickle ICE
-			"fingerprint:sha-256 " + fingerprint,
+		ProtocolVersion:  0,
+		Origin:           Origin{"-", newSessionID(), uint64(time.Now().Unix()), "IN", "IP4", "0.0.0.0"},
+		SessionName:      "-",
+		TimeDescriptions: []TimeDescription{{Timing{0, 0}, nil}},
+		Attributes: []Attribute{
+			// 	"Attribute(ice-options:trickle)", // TODO: implement trickle ICE
+			Attribute("fingerprint:sha-256 " + fingerprint),
```

Comment:
What do you think of making Attribute an interface, and having `AttributeKey` `AttributeKeyValue` would be nice if this just looked like `AttributeKeyValue("fingerprint", sha-256 " + fingerprint)` that is my biggest concern in the SDP code.

I don't want any string munging outside the SDP code (like the stuff in signaling.go)

Reply:
Yeah. I was thinking about exactly the same thing. Lets wrap up this run over the code first and then I will look into how to actually make what you are talking about. But generally speaking I am in agreement.